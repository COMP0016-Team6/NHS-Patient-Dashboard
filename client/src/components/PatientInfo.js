import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import { changedWeight } from "../state/action";
import TreatmentHistory from "./TreatmentHistory";
import { Link } from "react-router-dom";
import { changeWeight } from "../api/fetches";
import "react-toastify/dist/ReactToastify.css";
import { toast } from "react-toastify";
import { useStylesUser } from "../styles/styles";
import NavBar from "./Navbar";

toast.configure();

const PatientInfo = ({ logout }) => {
  const classes = useStylesUser();
  const dispatch = useDispatch();
  const treatmentPlan = useSelector(state => state.patientPlan);
  const isClinician = useSelector(state => state.isClinician);
  const patientInfo = useSelector(state => state.patientInfo);

  const { user_id, user_name, user_email, patient_gender, patient_dob, diagnostic_conclusion, weight } = patientInfo;

  const [isChangeWeight, setChangeWeight] = useState(false);
  const [newWeight, setWeight] = useState(weight);

  const changePatientWeight = async () => {
    try {
      const parseRes = await changeWeight({ user_id, newWeight });

      if (parseRes === "Success") {
        dispatch(changedWeight(newWeight));
        toast.success("Weight Change Successful");
        setChangeWeight(false);
      } else {
        toast.error(parseRes);
      }
    } catch (err) {
      console.error(err.message);
    }
  }
  
  function calcAge(dateString) {
    let YEARLEN = 31557600000;
    var birthday = +new Date(dateString);
    return ~~((Date.now() - birthday) / (YEARLEN));
  }

  return (
    <div className={classes.root}>
      <NavBar heading="" logout={logout}/>
      
      <div style={{width: "75%", justify: "center", alignItems: "center", margin: "auto"}}>
          {isClinician? (<Link to={`/dashboard/${user_id}`}> <button className="btn btn-primary mt-5">Back</button> </Link>)
            : (<Link to={`/dashboard`}> <button className="btn btn-primary mt-5">Back</button> </Link>)}

          <div className="mt-5">
            <h2 className="text-info mt-5">Patient Name: {user_name}</h2>
            <h2>Email Address: {user_email} </h2>
            <h2>Date of Birth: {new Date(patient_dob).toLocaleDateString()} ({calcAge(patient_dob)} years)</h2>
            <h2>Gender: {patient_gender}</h2>
            <h4>Diagnostic Conclusion: {diagnostic_conclusion}</h4>
            {!isChangeWeight? 
            <div>
            <div style={{display: "inline-block"}}><h4>Weight: {weight}</h4></div> <div style={{display: "inline-block", marginLeft: 10}}><button type="submit" className="btn btn-info" onClick={()=>setChangeWeight(true)}>Change Weight</button></div>
            </div>
            : <div>
                <h4> Weight:</h4>
                <input type="text" value={newWeight} placeholder="weight" onChange={e => setWeight(e.target.value)} className="form-control my-3"/>
                <button type="submit" className="btn btn-success" onClick={changePatientWeight} >Change Weight</button>
                <button type="submit" className="btn btn-danger ml-2" onClick={()=>{setWeight(weight); setChangeWeight(false)}}>Cancel</button>
              </div>
            }
            <h4>Treatment History: </h4>
            <div className="mb-5">
              {treatmentPlan.length === 0? null : 
              <TreatmentHistory treatmentPlan={treatmentPlan} />
              }
            </div>
          </div>
      </div>
    </div>
  )
}

export default PatientInfo;