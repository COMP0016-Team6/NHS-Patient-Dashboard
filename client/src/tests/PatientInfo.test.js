import React from "react";
import PatientInfo from "../components/PatientInfo";
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { patientWrapper } from "./util";

it ("PatientInfo snapshot matches", () => {
  const tree = render(patientWrapper(<BrowserRouter><PatientInfo /></BrowserRouter>));
  expect(tree).toMatchSnapshot();
});