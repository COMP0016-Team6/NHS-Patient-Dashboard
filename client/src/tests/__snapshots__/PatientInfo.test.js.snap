// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientInfo snapshot matches 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
          style="background: rgb(6, 107, 221);"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <h4
              class="makeStyles-title-2"
            >
              
               
              Test User Two
            </h4>
            <button
              class="btn btn-info"
            >
              Logout
            </button>
          </div>
        </header>
        <div
          style="width: 75%; align-items: center; margin: auto;"
        >
          <a
            href="/dashboard"
          >
             
            <button
              class="btn btn-primary mt-5"
            >
              Back
            </button>
             
          </a>
          <div
            class="mt-5"
          >
            <h2
              class="text-info mt-5"
            >
              Patient Name: 
              Test User Two
            </h2>
            <h2>
              Email Address: 
              test.user.two@gmail.com
               
            </h2>
            <h2>
              Date of Birth: 
              03/01/2000
               (
              21
               years)
            </h2>
            <h2>
              Gender: 
              Male
            </h2>
            <h4>
              Diagnostic Conclusion: 
              Lung Cancer I
            </h4>
            <div>
              <div
                style="display: inline-block;"
              >
                <h4>
                  Weight: 
                  70
                </h4>
              </div>
               
              <div
                style="display: inline-block; margin-left: 10px;"
              >
                <button
                  class="btn btn-info"
                  type="submit"
                >
                  Change Weight
                </button>
              </div>
            </div>
            <h4>
              Treatment History: 
            </h4>
            <div
              class="mb-5"
            >
              <div
                class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
              >
                <table
                  aria-label="simple table"
                  class="MuiTable-root makeStyles-table-5"
                >
                  <thead
                    class="MuiTableHead-root"
                  >
                    <tr
                      class="MuiTableRow-root MuiTableRow-head"
                    >
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                        scope="col"
                      >
                        Treatment Plan Description
                      </th>
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                        scope="col"
                      >
                        Target Fluid (mL)
                      </th>
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                        scope="col"
                      >
                        Target Energy (kcal)
                      </th>
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                        scope="col"
                      >
                        Modified On
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="MuiTableBody-root"
                  >
                    <tr
                      class="MuiTableRow-root"
                      style="overflow: scroll;"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                      >
                        test
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                      >
                        1.5
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                      >
                        2.5
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                      >
                        2020-05-05 20:20:20
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
        style="background: rgb(6, 107, 221);"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <h4
            class="makeStyles-title-2"
          >
            
             
            Test User Two
          </h4>
          <button
            class="btn btn-info"
          >
            Logout
          </button>
        </div>
      </header>
      <div
        style="width: 75%; align-items: center; margin: auto;"
      >
        <a
          href="/dashboard"
        >
           
          <button
            class="btn btn-primary mt-5"
          >
            Back
          </button>
           
        </a>
        <div
          class="mt-5"
        >
          <h2
            class="text-info mt-5"
          >
            Patient Name: 
            Test User Two
          </h2>
          <h2>
            Email Address: 
            test.user.two@gmail.com
             
          </h2>
          <h2>
            Date of Birth: 
            03/01/2000
             (
            21
             years)
          </h2>
          <h2>
            Gender: 
            Male
          </h2>
          <h4>
            Diagnostic Conclusion: 
            Lung Cancer I
          </h4>
          <div>
            <div
              style="display: inline-block;"
            >
              <h4>
                Weight: 
                70
              </h4>
            </div>
             
            <div
              style="display: inline-block; margin-left: 10px;"
            >
              <button
                class="btn btn-info"
                type="submit"
              >
                Change Weight
              </button>
            </div>
          </div>
          <h4>
            Treatment History: 
          </h4>
          <div
            class="mb-5"
          >
            <div
              class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <table
                aria-label="simple table"
                class="MuiTable-root makeStyles-table-5"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                      scope="col"
                    >
                      Treatment Plan Description
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                      scope="col"
                    >
                      Target Fluid (mL)
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                      scope="col"
                    >
                      Target Energy (kcal)
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                      scope="col"
                    >
                      Modified On
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root"
                    style="overflow: scroll;"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                    >
                      test
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                    >
                      1.5
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                    >
                      2.5
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                    >
                      2020-05-05 20:20:20
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
